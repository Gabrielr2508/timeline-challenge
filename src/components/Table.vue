<template>
  <table>
    <col width="80%">
    <col width="20%">
    <tr class="border-bottom">
      <th
        v-for="field in fields"
        :key="field.key"
        :class="field.class"
        class="font-weight-semibold"
      >
        {{ field.name }}
      </th>
    </tr>
    <tr
      v-for="(row, index) in data"
      :key="index"
      :class="{ 'border-bottom': index !== data.length - 1 }"
    >
      <td
        v-for="field in fields"
        :key="field.key"
        :class="field.class"
      >
        {{ row[field.key] }}
      </td>
    </tr>
  </table>
</template>

<script>
export default {
  props: {
    fields: {
      required: true,
      type: Array,
      default: () => ([]),
    },
    data: {
      required: true,
      type: Array,
      default: () => ([]),
    },
  },
};
</script>

<style lang="scss" scoped>
  table {
    width: 100%;
  }
  th {
    padding: 0 0 0.6em 0;
    margin: 0 0 1.5em 0;
  }
  td {
    padding: 0.6em 0 0.6em 0;
  }
</style>
