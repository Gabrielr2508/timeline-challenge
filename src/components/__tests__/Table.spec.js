import { shallowMount } from '@vue/test-utils';
import Table from '../Table.vue';


import fieldsConfig from '../../configs/timelineTable';

const data = [
  {
    price: 'R$ 100,00',
    product: 'Camisa Azul',
  },
  {
    price: 'R$ 150,00',
    product: 'CalÃ§a Rosa',
  },
];

describe('Table.vue', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      fields: fieldsConfig.fields,
      data,
    };

    wrapper = shallowMount(Table, {
      propsData: { ...props },
    });
  });


  test('renders IconAndText correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders one <th> for each field', () => {
    expect(wrapper.findAll('th').length).toBe(2);
  });

  test('renders one <td> for each field data', () => {
    expect(wrapper.findAll('td').length).toBe(4);
  });
});
