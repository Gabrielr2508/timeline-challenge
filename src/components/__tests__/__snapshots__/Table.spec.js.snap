// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue renders IconAndText correctly 1`] = `
<table>
  <tr
    class="border-bottom"
  >
    <th
      class="font-weight-semibold text-md w-80"
    >
      
      Produto
    
    </th>
    <th
      class="font-weight-semibold text-md w-20"
    >
      
      Preço
    
    </th>
  </tr>
   
  <tr
    class="border-bottom"
  >
    <td
      class="text-md w-80"
    >
      
      Camisa Azul
    
    </td>
    <td
      class="text-md w-20"
    >
      
      R$ 100,00
    
    </td>
  </tr>
  <tr
    class=""
  >
    <td
      class="text-md w-80"
    >
      
      Calça Rosa
    
    </td>
    <td
      class="text-md w-20"
    >
      
      R$ 150,00
    
    </td>
  </tr>
</table>
`;
